engine/collection/epochmgr/engine_test.go

```diff
@@ -178,14 +195,46 @@ func (suite *Suite) TestRestartInSetupPhase() {
 		}).Once()
 
 	// start up the engine
-	<-suite.engine.Ready()
+	unittest.AssertClosesBefore(suite.T(), suite.engine.Ready(), time.Second)
 	suite.Assert().Eventually(func() bool {
 		return called
 	}, time.Second, time.Millisecond)
 
 	suite.voter.AssertExpectations(suite.T())
 }
 
+// When a collection node joins the network at an epoch boundary, they must
+// start running during the EpochSetup phase in the epoch before they become
+// a staked member so they submit their cluster QC vote.
+//
+// These nodes must kick off the root QC voter but should not attempt to
+// participate in cluster consensus in the current epoch.
+func (suite *Suite) TestStartAsUnstakedNode() {
+	suite.MockAsUnstakedNode()
+
+	// we are in setup phase
+	suite.snap.On("Phase").Return(flow.EpochPhaseSetup, nil)
+
+	// should call voter with next epoch
+	var called bool
```

Comment:
Just to add more readability, I would suggest to use wait group here, then instead of 

```
	suite.Assert().Eventually(func() bool {
		return called
	}, time.Second, time.Millisecond)
```

you may use:

```
unittest.AssertReturnsBefore(suite.T(), wg.Wait, 1 * time.Second)
```